frontend/src/common/SortedCollection.ts

```diff
@@ -97,27 +97,44 @@ export namespace SortedCollection {
   export type StateRef = Opaque<number, 'SortedCollection.StateRef'>;
 }
 
+interface Focus {
+  start: number;
+  end: number;
+}
+
 export class SortedCollection<Item extends { id: number }> {
+  // Comparator function used to sort the collection
   private compare: Compare<Item>;
+  // Mapping item `id` to the `Item`, this uses array as a structure with
+  // the assumption that `id`s provided are increments from `0`, and that
+  // vacant `id`s will be re-used in the future.
   private map = Array<Maybe<Item>>();
+  // Actual sorted list of `Item`s.
   private list = Array<Item>();
+  // Internal tracker for changes, this number increments whenever the
+  // order of the **focused** elements in the collection changes
   private changeRef = 0;
+  // Marks the range of indicies that are focused for tracking.
+  // **Note:** `start` is inclusive, while `end` is exclusive (much like
+  // `Array.slice()`).
+  private focus: Focus = { start: 0, end: 0 };
 
   constructor(compare: Compare<Item>) {
     this.compare = compare;
   }
 
+  // Set a new focus within which order changes bump `changeRef`
```

Comment:
```suggestion
  // Set a new `Focus`. Any changes to the order of items within the `Focus` will trigger an increase to `changeRef`, which in turn triggers a re-render.
```

…maybe?

Reply:
Updated the comment a bit (it doesn't technically trigger anything, plus the abstraction of this particular class knows nothing about the rendering pipeline), and moved it next to `ref()` and `hasChangedSince()` so that intent is clearer.